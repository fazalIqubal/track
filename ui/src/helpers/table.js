export const alternateRowBy = (alternatingValue: number, alternateEven: boolean = true) => {
  return alternateEven
    ? (alternatingValue % 2 === 0) ? '' : 'alternate'
    : (alternatingValue % 2 !== 0) ? '' : 'alternate';
};

export const alternateRowByIndex = (_record: any, index: number) => {
  return alternateRowBy(index);
};
