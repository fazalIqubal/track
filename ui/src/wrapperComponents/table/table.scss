#ui_block{
    text-align: center;
}
.ant-table-thead > tr > th {
    font-weight: 700;
}