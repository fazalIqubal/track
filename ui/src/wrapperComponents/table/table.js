import React from 'react'
import {
    Table, Pagination
} from 'antd';
import './table.scss'
import PageLoader from '../loader/loader'

export class TableComponent extends React.Component {
    constructor() {
        super();
        this.state = {
            searchText: '',
            searchedColumn: '',
            filteredInfo: {},
            sortedInfo: {},
            data: [],
            columns: [],
            pageSizeOptions: ['5', '10', '20', '30'],
            defaultPageSize: 5,
        };
    }

    componentWillReceiveProps(nextProps) {
        if (nextProps.dataSource) {
            this.listMaper((this.state.columns || this.props.columns), nextProps.dataSource)
        }
    }

    componentDidMount() {
        this.listMaper(this.props.columns, this.props.dataSource)
    }

    listMaper = (columns, dataSource) => {
        columns.map(column => {
            if (column.isFilters == true && dataSource) {
                column.filters = dataSource.map(data => {
                    var filter = { text: data[column.dataIndex], value: data[column.dataIndex] }
                    return filter
                });
                column.onFilter = (value, record) => this.onFilterHandeler(column.dataIndex, value, record)
                var obj = {};
                for (var i = 0, len = column.filters.length; i < len; i++) {
                    obj[column.filters[i]['text']] = column.filters[i];
                }
                column.filters = new Array();
                for (var key in obj) {
                    column.filters.push(obj[key]);
                }
            }

            console.log(column.filters)
        })
        this.setState({ columns: columns, data: dataSource })
    }
    onFilterHandeler = (columnName, value, record) => {
        return ((record[columnName]).toString()).includes(value.toString())
    }

    handleChange = (pagination, filters, sorter) => {
        console.log('Various parameters', pagination, filters, sorter);
        this.setState({
            filteredInfo: filters,
            sortedInfo: sorter,
        });
    };

    clearFilters = () => {
        this.setState({ filteredInfo: {} });
    };

    clearAll = () => {
        this.setState({
            filteredInfo: {},
            sortedInfo: {},
        });
    };
    render() {
        let length = (this.state.data && this.state.data.length) || 0
        return (
            <Table columns={this.state.columns}
                dataSource={this.state.data}
                onChange={this.handleChange}
                pagination={{ defaultPageSize: this.state.defaultPageSize, showSizeChanger: true, pageSizeOptions: this.state.pageSizeOptions, total: length, hideOnSinglePage: this.props.hide }}
            >
            </Table>
        )
    }
}