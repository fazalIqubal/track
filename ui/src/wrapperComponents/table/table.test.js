import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { TableComponent } from './table';

Enzyme.configure({ adapter: new Adapter() })

const columns = [
    {
        title: 'Issue ID',
        dataIndex: 'issueId',
        key: 'issueId',
        render: text => <a>{text}</a>,
        className: 'columnTitle'
    },
    {
        title: 'Title',
        dataIndex: 'title',
        key: 'title',
        isFilters: true,
        className: 'columnTitle'

    },
    {
        title: 'Issue Status',
        dataIndex: 'issueStatus',
        key: 'issueStatus',
        isFilters: true,
        className: 'columnTitle'
    },
    {
        title: 'Priority',
        key: 'tags',
        dataIndex: 'tags',

        render: tags => (
            <>
                {tags.map(tag => {
                    let color = tag == 'High' ? 'red' : tag == 'Medium' ? 'orange' : 'green'
                    return (
                        <Tag color={color} key={tag}>
                            {tag.toUpperCase()}
                        </Tag>
                    );
                })}
            </>
        ),
        className: 'columnTitle'
    },  
    {
        title: 'Issue Type',
        dataIndex: 'issueType',
        key: 'issueType',
        className: 'columnTitle'
    },
    {
        title: 'Reporter',
        dataIndex: 'issueReporter',
        key: 'issueReporter',
        className: 'columnTitle'
    },
    {
        title: 'Assigned To',
        dataIndex: 'name',
        key: 'name',
        className: 'columnTitle'
    },
]
const dataSet = [
    {
        key: '1',
        issueId: '001',
        title: 'Issue #1',
        issueStatus: 'Open',
        tags: ['Medium'],
        issueType: 'Defect',
        issueReporter: 'Chetan Dhami',
        name: 'John Brown',
    },
    {
        key: '2',
        issueId: '002',
        title: 'Issue #2',
        issueStatus: 'In-Progress',
        tags: ['Low'],
        issueType: 'UI Fix',
        issueReporter: 'Yogesh Dhami',
        name: 'John Brown',
    },
    {
        issueId: '001',
        title: 'Issue #3',
        issueStatus: 'Open',
        tags: ['High'],
        issueType: 'New Feature',
        issueReporter: 'Chetan Dhami',
        name: 'John Brown',
    },
]
let output
describe('Table component test case', () => {
    beforeEach(() => {
        output = shallow(
            <TableComponent columns={columns} dataSource={dataSet} />
        );
    })
    afterEach(() => {
        output = null;
    });

    it('should render correctly', () => {
        expect(shallowToJson(output)).toMatchSnapshot();
    })

})