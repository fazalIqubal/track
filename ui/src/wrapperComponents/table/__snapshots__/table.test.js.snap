// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component test case should render correctly 1`] = `
<Table
  columns={
    Array [
      Object {
        "className": "columnTitle",
        "dataIndex": "issueId",
        "key": "issueId",
        "render": [Function],
        "title": "Issue ID",
      },
      Object {
        "className": "columnTitle",
        "dataIndex": "title",
        "filters": Array [
          Object {
            "text": "Issue #1",
            "value": "Issue #1",
          },
          Object {
            "text": "Issue #2",
            "value": "Issue #2",
          },
          Object {
            "text": "Issue #3",
            "value": "Issue #3",
          },
        ],
        "isFilters": true,
        "key": "title",
        "onFilter": [Function],
        "title": "Title",
      },
      Object {
        "className": "columnTitle",
        "dataIndex": "issueStatus",
        "filters": Array [
          Object {
            "text": "Open",
            "value": "Open",
          },
          Object {
            "text": "In-Progress",
            "value": "In-Progress",
          },
          Object {
            "text": "Open",
            "value": "Open",
          },
        ],
        "isFilters": true,
        "key": "issueStatus",
        "onFilter": [Function],
        "title": "Issue Status",
      },
      Object {
        "className": "columnTitle",
        "dataIndex": "tags",
        "key": "tags",
        "render": [Function],
        "title": "Priority",
      },
      Object {
        "className": "columnTitle",
        "dataIndex": "issueType",
        "key": "issueType",
        "title": "Issue Type",
      },
      Object {
        "className": "columnTitle",
        "dataIndex": "issueReporter",
        "key": "issueReporter",
        "title": "Reporter",
      },
      Object {
        "className": "columnTitle",
        "dataIndex": "name",
        "key": "name",
        "title": "Assigned To",
      },
    ]
  }
  dataSource={
    Array [
      Object {
        "issueId": "001",
        "issueReporter": "Chetan Dhami",
        "issueStatus": "Open",
        "issueType": "Defect",
        "key": "1",
        "name": "John Brown",
        "tags": Array [
          "Medium",
        ],
        "title": "Issue #1",
      },
      Object {
        "issueId": "002",
        "issueReporter": "Yogesh Dhami",
        "issueStatus": "In-Progress",
        "issueType": "UI Fix",
        "key": "2",
        "name": "John Brown",
        "tags": Array [
          "Low",
        ],
        "title": "Issue #2",
      },
      Object {
        "issueId": "001",
        "issueReporter": "Chetan Dhami",
        "issueStatus": "Open",
        "issueType": "New Feature",
        "name": "John Brown",
        "tags": Array [
          "High",
        ],
        "title": "Issue #3",
      },
    ]
  }
  onChange={[Function]}
  pagination={
    Object {
      "hideOnSinglePage": true,
    }
  }
  rowKey="key"
/>
`;
